spyder/api/plugins.py

```diff
@@ -533,3 +550,697 @@ def set_ancestor(self, ancestor):
+        """
+        self.sig_option_changed.disconnect()
+        self._conf.unregister_plugin()
+        self._is_registered = False
+        self._is_compatible = None
+        self._widget = None
+
+    def _register_shortcut(self, qaction, context, name,
+                           add_shortcut_to_tip=False):
+        """
+        Register a shortcut associated to a QAction or a QShortcut to
+        Spyder main application.
+
+        Parameters
+        ----------
+        qaction: QAction
+            QAction to register the shortcut for.
+        context: str
+            Name of the plugin this shortcut applies to. For instance,
+            if you pass 'Editor' as context, the shortcut will only
+            work when the editor is focused.
+            Note: You can use '_' if you want the shortcut to be work
+            for the entire application.
+        name: str
+            Name of the action the shortcut refers to (e.g. 'Debug
+            exit').  TODO: This could be overriden by the actual
+            action `name`(not the text!)
+        add_shortcut_to_tip: bool
+            If True, the shortcut is added to the action's tooltip.
+            This is useful if the action is added to a toolbar and
+            users hover it to see what it does.
+        """
+        self.sig_shortcut_registration_requested.emit(
+            qaction,
+            context,
+            name,
+            add_shortcut_to_tip,
+        )
+
+    # --- API: available methods ---------------------------------------------
+    # ------------------------------------------------------------------------
+    def create_action(self, name, text, icon=None, tip=None, toggled=None,
+                      triggered=None, data=None, menu_role=None,
+                      context=Qt.WindowShortcut, shortcut_context=None,
+                      expose=True, disable_if_not_visible=True, parent=None):
+        """
+        name: str
+            unique identifiable name for the action
+        text: str
+           Localized text for the action
+        ...
+        """
+        if self._actions is None:
+            self._actions = OrderedDict()
+
+        if name in self._actions:
+            raise Exception('Action name "{}" already in use!'.format(name))
+
+        if parent is None:
+            parent = self
+
+        action = create_action(
+            parent,
+            text=text,
+            icon=icon,
+            tip=tip,
+            toggled=toggled,
+            triggered=triggered,
+            data=data,
+            menurole=menu_role,
+            context=context,
+        )
+        action.name = name
+        action.shortcut_context = shortcut_context
+        action.expose = expose
+        action.disable_if_not_visible = disable_if_not_visible
+        self._actions[name] = action
+        return action
+
+    def get_path(self):
+        """
+        Return the plugin system path.
+        """
+        return self._plugin_path
```

Comment:
We could, but since PyQt does not follow this pattern a lot, it feels more natural to provide similar getters on our API. Is more of a consistency issue.